### ISSUE TITLE
v2.1.0 ignores all controllers

### ISSUE THREAD

**Issue by bbapnl** [danadesrosiers/silex-annotation-provider#30]
Hello,

I have just updated to version 2.1.0 and all my routes are now ignored because of a combination of `AnnotationService::getFiles()` and `AnnotationService::registerControllers()` / `AnnotationService::prefixMatchesUri()`.

`AnnotationService::getFiles()` builds an array which is indexed with both prefixes and indexes, then it is used by `AnnotationService::registerControllers()` in a `foreach($controllers as $prefix => $controllerNames)`. 
When my controller doesn't have a prefix, the position in the array is used as a prefix `/0`, `/1` and so on, for example:

```
string(48) "Controller\CommonController"
string(7) "$prefix"
int(1)
string(2) "/1"
```

When my controller does have a prefix, it is re-prefixed by "/" which make the `strpos` function fail, for example:

```
string(57) "Modules\Administration\ApiController" (I am using multiple controller directories)
string(7) "$prefix"
string(19) "/administration/api"
string(20) "//administration/api"
```

Here's how I register the service provider

```php
<?php
$this->register(
  new AnnotationServiceProvider(),
  [
    'annot.base_uri'      => '/',
    'annot.cache'         => new ArrayCache(),
    'annot.controllerDir' => [
      __DIR__.'/Controller/',
      __DIR__.'/Modules/',
    ],
  ]
);
```

I have, for now, regressed to version 2.0.1 so as to keep developing but I thought I would mention it here, be it an error on my part because I simply run `composer update` or a code error. I am running PHP 7.2 on Apache 2.4 on Windows 10 x64.

Thank you,
Bruno Buiret

**Comment by danadesrosiers** [comment]
Thank you for reporting the problem.  I thought I had that case covered, but maybe I only did in v3.  I will work on it as soon as I can.  In the mean time, you should be able continue to use 2.1.  The 'annot.base_uri' is optional.  If you leave it out when you register, it should ignore the fast register logic and behave exactly like 2.0.1.

**Comment by bbapnl** [comment]
You're welcome! I have tried to re-upgrade to 2.1 and I tested both with and without `'annot.base_uri' => '/'`, both times the fast register logic applied.

I matched the controller linked to `/contents/library` without the `annot.base_uri` but not with it. I have made several `var_dump()` if you need more data about what is being tested. Moreover, when matching my controller, I used the `UrlGenerator` inside it to generate an URL to another controller that isn't prefixed by `/contents/library` and, since fast register logic applied, I couldn't generate this URL because the routes wasn't registered.

I'm so sorry, I feel like I'm only critizing your work but I was happy to see you had pushed more code! =|

**Comment by danadesrosiers** [comment]
Don't be sorry, I'm not offended by your bug report.  I am, however, deeply embarrassed to have released such broken code.  I must have been too focused on v3 and didn't realize I didn't finish v2.1.  I'm working on a fix and will have a new release by the end of the weekend.

**Comment by danadesrosiers** [comment]
This issue is fixed by #31.  v2.1.1 has been created.

Fast register is now always enabled.  The base_uri thing was not necessary, I was just confused.  It should be good now.  Please open another ticket if you encounter any more problems.
